python/tests/test_table.py

```diff
@@ -79,3 +79,35 @@ def test_add(db):
             .to_table()
         )
         assert expected == tbl
+
+
+def test_add_items(db):
+    table = LanceTable.create(
+        db,
+        "test",
+        data=[
+            {"vector": [3.1, 4.1], "item": "foo", "price": 10.0},
+            {"vector": [5.9, 26.5], "item": "bar", "price": 20.0},
+        ],
+    )
+
+    # table = LanceTable(db, "test")
+    assert len(table) == 2
+
+    count = table.add([{"vector": [6.3, 100.5], "item": "new", "price": 30.0}])
+    assert count == 3
+    #assert len(table) == 3 #FAILS! len(table) == 2, since add creates a new ds
```

Comment:
the table instance holds a reference to a lance dataset right? 
the call to `lance.write_dataset(mode="append")` gives a LanceDataset as output. the table should update it's reference to use that new instance